### PR TITLE
XMPP: build_identifier can now build Rooms

### LINKED ISSUE #717: xmpp channel message failing
`self.send(self.build_identifier('dev@conference.jabber.domain.tld'), "test",)`  fails with 

`WARNING errbot.backends.xmpp Received error message: <message to="devbot@jabber.domain.tld/9oednv6no3" type="error" xml:lang="en" from="dev@conference.jabber.domain.tld"><body>test</body><error code="400" type="modify"><bad-request xmlns="urn:ietf:params:xml:ns:xmpp-stanzas" /></error></message>`

Tested with errbot 4.0.3 on openfire 4.0.2

### PR DESCRIPTION
**Pull request by gbin**
but also RoomOccupants by querying the XMPP discovery service.

This fixes #717

### CODE REVIEW

File: errbot/backends/xmpp.py [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import logging
 import sys
+from functools import lru_cache
```
Comment (zoni) [inline errbot/backends/xmpp.py]: Python 2 compatibility requires:

``` python
try:
    from functools import lru_cache
except ImportError:
    from backports.functools_lru_cache import lru_cache
```

File: errbot/backends/xmpp.py [hunk 2]
```diff
@@ -31,6 +32,9 @@
     """)
     sys.exit(-1)
 
+# LRU to cache the JID queries.
+IDENTIFIERS_LRU = 1000
```
Comment (zoni) [inline errbot/backends/xmpp.py]: According to the documentation for `lru_cache`, it performs better when using a power of 2, so I suggest changing this to `1024`

File: errbot/backends/xmpp.py (hunk 1 above)
Comment (gbin) [inline errbot/backends/xmpp.py]: done

File: errbot/backends/xmpp.py (hunk 2 above)
Comment (gbin) [inline errbot/backends/xmpp.py]: done

**Comment by zoni** [comment]
Code-wise this looks OK, but I don't have an XMPP setup handy to test this out against a real server right now. I'll try to do that over the weekend.

**Comment by gbin** [comment]
cool, I made a last minute change, I forgot to reuse the _build_room_occupant and _build_person in build_identifier itself.

**Comment by gbin** [comment]
Thanks for the review, I tested it on blah.im but I hope I did not hit any specificity for this server.

**Pull request merged**